### PR TITLE
lightning: merge configuration for preprocess duplicate detection and post-import conflict detection

### PR DESCRIPTION
**Pull request by lyzx2001**
### What problem does this PR solve?

Issue Number: ref #51036 

Problem Summary:

### What changed and how does it work?

Merge preprocess duplicate detection and post-import conflict detection.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
The configuration for lightning preprocess duplicate detection and post-import conflict detection has been merged to `conflict.strategy`.
```

### CODE REVIEW

File: br/pkg/lightning/backend/local/duplicate.go [hunk 1]
```diff
@@ -809,7 +809,7 @@ func (m *DupeDetector) buildLocalDupTasks(dupDB *pebble.DB, keyAdapter common.Ke
 }
 
 // CollectDuplicateRowsFromDupDB collects duplicates from the duplicate DB and records all duplicate row info into errorMgr.
-func (m *DupeDetector) CollectDuplicateRowsFromDupDB(ctx context.Context, dupDB *pebble.DB, keyAdapter common.KeyAdapter, algorithm config.DuplicateResolutionAlgorithm) error {
+func (m *DupeDetector) CollectDuplicateRowsFromDupDB(ctx context.Context, dupDB *pebble.DB, keyAdapter common.KeyAdapter, algorithm string) error {
```
Comment (Benjamin2037) [inline br/pkg/lightning/backend/local/duplicate.go:812]: can we use an enum type to represent the algorithm of duplicate record handling mode？

Comment (lyzx2001) [reply br/pkg/lightning/backend/local/duplicate.go]: Originally `conflict.strategy` was in the type of `string`, so I maintained it in the old code. However, using enum type indeed is clearer, so I have changed the type to enum.

File: br/pkg/lightning/backend/local/local.go [hunk 2]
```diff
@@ -440,8 +440,8 @@ func NewBackendConfig(cfg *config.Config, maxOpenFiles int, keyspaceName, resour
 		MemTableSize:                int(cfg.TikvImporter.EngineMemCacheSize),
 		LocalWriterMemCacheSize:     int64(cfg.TikvImporter.LocalWriterMemCacheSize),
 		ShouldCheckTiKV:             cfg.App.CheckRequirements,
-		DupeDetectEnabled:           cfg.TikvImporter.DuplicateResolution != config.DupeResAlgNone,
-		DuplicateDetectOpt:          common.DupDetectOpt{ReportErrOnDup: cfg.TikvImporter.DuplicateResolution == config.DupeResAlgErr},
+		DupeDetectEnabled:           cfg.TikvImporter.Backend == config.BackendLocal && cfg.Conflict.Strategy != config.NoneOnDup,
```
[the remark this answers is not in the document]
Comment (lyzx2001) [reply br/pkg/lightning/backend/local/local.go]: right, modified

File: pkg/ddl/ingest/config.go [hunk 3]
```diff
@@ -63,10 +63,12 @@ func genConfig(ctx context.Context, memRoot MemRoot, jobID int64, unique bool, r
 	adjustImportMemory(ctx, memRoot, cfg)
 	cfg.Checkpoint.Enable = true
 	if unique {
-		cfg.TikvImporter.DuplicateResolution = lightning.DupeResAlgErr
+		if cfg.TikvImporter.Backend == lightning.BackendLocal {
```
Comment (D3Hunter) [inline pkg/ddl/ingest/config.go:66]: add-index always use local backend, maybe remove this. below too

File: br/tests/lightning_foreign_key/local-config.toml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+[tikv-importer]
+# Set on-duplicate=error to force using insert statement to write data.
```
Comment (D3Hunter) [inline br/tests/lightning_foreign_key/local-config.toml:2]:
```suggestion
```

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: pkg/ddl/ingest/config.go (hunk 3 above)
Comment (lyzx2001) [reply pkg/ddl/ingest/config.go]: modified

**Review by D3Hunter — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Review by 3pointer — APPROVED** [review]

**Pull request merged**